FrontEnd/src/HomePage/MainPage.jsx

```diff
@@ -32,36 +67,41 @@ function MainPage({ userUID, userData, socket }) {
   return (
     <div className={`homePageList ${theme}`}>
       <div className="categorySection">
-        <CategoryList />
+        <CategoryList filterPosts={filterPosts} />
       </div>
-      <div className="feedSection">
-        {isLoading ? (
-          <p>Loading...</p>
-        ) : userFeed.length === 0 ? (
-          <p>No posts found.</p>
-        ) : (
-          userFeed.map((post, index) => (
-            <div
-              key={index}
-              className="postFeed"
-              onClick={() => toggleModal(post)}
-            >
-              <PostsPreview post={post} />
+      <div className="bodyArea">
+        <div className="searchArea">
+          <SearchBar filterPosts={filterPosts} />
+        </div>
+        <div className="feedSection">
+          {isLoading ? (
+            <p>Loading...</p>
```

Comment:
For loading states would be great to show some loading spinner. It can be a reusable component that you show in all the places where stuff is loading.